src/App.js

```diff
@@ -1,31 +1,51 @@
-import React from 'react'
+import React, { useContext } from 'react'
+import { useHistory } from 'react-router-dom'
+
+import UserProvider, { UserContext } from "./providers/UserProvider";
 import { BrowserRouter as Router, Switch, Route } from 'react-router-dom'
+
 import PacklistPage from './Pages/Packlist/PacklistPage'
-import EventPage from './Pages/EventMain/EventPage'
+import UserPage from './Pages/UserPage/UserPage'
 import EventForm from './Pages/EventForm/EventForm'
-import SettingsPage from './Pages/SettingsPage/SettingsPage.js'
+import SettingsPage from './Pages/SettingsPage/SettingsPage'
+import StartPage from './Pages/StartPage/StartPage'
+import GuestPage from './Pages/GuestPage/GuestPage'
 import PacklistForm from './Pages/PacklistForm/PacklistForm'
 
 export default function App() {
+  const history = useHistory()
+  const user = useContext(UserContext);
+  console.log('app user', user, !!user);
   return (
-    <Router>
-      <Switch>
-        <Route path="/settings">
-          <SettingsPage />
-        </Route>
-        <Route path="/packlist/:packlistName?">
-          <PacklistPage />
-        </Route>
-        <Route path="/packlistform/:packlistId?">
-          <PacklistForm />
-        </Route>
-        <Route path="/eventform/:eventId?">
-          <EventForm />
-        </Route>
-        <Route path="/">
-          <EventPage />
-        </Route>
-      </Switch>
-    </Router>
+    <UserProvider>
```

Comment:
I saw the GuardedRoute component but cant find it here.
Did you implemented it elsewhere?

Reply:
I tried to implement it, but it wasnt working, so I need another solution for a guarded route. At the moment, thhe user can use the known url for users to get to the page, without an login, so I try to fix this qith an guarded route

Reply:
Okay, i see :-1: 
I implemented it also in february 2020. Maybe this can help?

https://gitlab.com/leonp5/whfm/-/blob/master/client/src/App.js

https://gitlab.com/leonp5/whfm/-/blob/master/client/src/components/ProtectedRoute.js

(Dont worry about the outcommented line. Login and protected route is working but the site has no registration process)

Reply:
Thanks for the feedback, I will merge this login pull request and open an new issue for the guarded route :) to fix it